internal/service/cloudmssql/cloud_mssql.go

```diff
@@ -0,0 +1,407 @@
+package cloudmssql
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/NaverCloudPlatform/ncloud-sdk-go-v2/ncloud"
+	"github.com/NaverCloudPlatform/ncloud-sdk-go-v2/services/vmssql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	. "github.com/terraform-providers/terraform-provider-ncloud/internal/common"
+	"github.com/terraform-providers/terraform-provider-ncloud/internal/conn"
+	. "github.com/terraform-providers/terraform-provider-ncloud/internal/verify"
+)
+
+func ResourceNcloudMssql() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNcloudMssqlCreate,
+		ReadContext:   resourceNcloudMssqlRead,
+		DeleteContext: resourceNcloudMssqlDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(conn.DefaultCreateTimeout),
+			Delete: schema.DefaultTimeout(3 * conn.DefaultTimeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vpc_no": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "VPC Number of Cloud DB for MSSQL instance.",
+				ForceNew:    true,
+			},
+			"subnet_no": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Subnet Number of Cloud DB for MSSQL instance.",
+				ForceNew:    true,
+			},
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateDiagFunc: ToDiagFunc(validation.All(
+					validation.StringLenBetween(3, 15),
+					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9\-가-힣]+$`), "Composed of alphabets, numbers, korean, hyphen (-)."),
+				)),
```

Comment:
과거 `ValidateFunc` -> `ValidateDiagFunc` 으로 변환해주는 helper가 sdk에서 제공되지 않았던 상태였었어서 저희 내부에 변환해주는 코드를 카피해서 사용했었는데요, 해당 함수가 2.4.0[#ref](https://github.com/hashicorp/terraform-plugin-sdk/issues/534) 버전 부터  SDK에 추가되었습니다. 아래와 같이 sdk 버전으로 사용하면 좋을것 같습니다~ 추후 다른 리소스들에서도 리펙토링 대상이겠네요!

```suggestion
				ValidateDiagFunc: validation.ToDiagFunc(validation.All(
					validation.StringLenBetween(3, 15),
					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9\-가-힣]+$`), "Composed of alphabets, numbers, korean, hyphen (-)."),
				)),
```